File: appjava/src/main/java/io/firekast/appjava/PlayerFragment.java
package io.firekast.appjava;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.TextUtils;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import io.firekast.FKError;
import io.firekast.FKPlayer;
import io.firekast.FKPlayerView;
import io.firekast.FKStream;

/**
 * Created by Francois Rouault on 01/11/2016.
 */
public class PlayerFragment extends Fragment implements FKPlayer.Callback, View.OnClickListener {

    private static final String TAG = PlayerFragment.class.getSimpleName();

    private EditText mEditTextUrl;
    private FKPlayerView mPlayerView;
    private TextView mTextViewState;

    private ProgressDialog mLoading;
    private FKPlayer mPlayer;

    public static PlayerFragment newInstance() {
        return new PlayerFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_player, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        mEditTextUrl = view.findViewById(R.id.editText_url);
        mPlayerView = view.findViewById(R.id.playerView);
        mTextViewState = view.findViewById(R.id.textViewState);
        view.findViewById(R.id.button_get_time).setOnClickListener(this);
        view.findViewById(R.id.button_play).setOnClickListener(this);
        view.findViewById(R.id.pauseButton).setOnClickListener(this);
        view.findViewById(R.id.resumeButton).setOnClickListener(this);

        mPlayer = mPlayerView.getPlayer();
        mPlayer.setCallback(this);
        mPlayer.setShowPlaybackControls(true);
        mEditTextUrl.setHint("Stream id (ex. H1N_D8eex)");
        mEditTextUrl.setText(App.latestStreamId);
        mTextViewState.setText("" + mPlayer.getState());
    }

    @Override
    public void onPause() {
        super.onPause();
        mPlayer.release();
    }

    public void showLoading() {
        mLoading = ProgressDialog.show(getContext(), null, "Loading stream...");
    }

    public void hideLoading() {
        if (mLoading == null) {
            return;
        }
        mLoading.dismiss();
    }

    public static String convertSecondsToHMmSs(long seconds) {
        long s = seconds % 60;
        long m = (seconds / 60) % 60;
        long h = (seconds / (60 * 60)) % 24;
        return String.format("%d:%02d:%02d", h, m, s);
    }

    // ------
    // ON CLICK
    // ------

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_play:
                onClickPlay();
                break;
            case R.id.button_get_time:
                onClickGetTime();
                break;
            case R.id.resumeButton:
                mPlayer.resume();
                break;
            case R.id.pauseButton:
                mPlayer.pause();
                break;

        }
    }

    void onClickGetTime() {
        long position = mPlayer.getCurrentPosition() / 1000L;
        String toHMmSs = convertSecondsToHMmSs(position);
        Toast.makeText(this.getContext(), "Position: " + toHMmSs, Toast.LENGTH_SHORT).show();
    }

    void onClickPlay() {
        String streamId = mEditTextUrl.getText().toString().trim();
        FKStream current = mPlayer.getCurrentStream();
        showLoading();
        if (TextUtils.isEmpty(streamId)) {
            mEditTextUrl.setError("Need a StreamID");
        } else if (current == null || !current.getId().equals(streamId)) {
            FKStream stream = FKStream.newEmptyInstance(streamId);
            mPlayer.play(stream);
        } else {
            mPlayer.play(current);
        }
    }

    // --------
    // FKPLayer Callback
    // --------

    @Override
    public void onPlayerWillPlay(@NonNull FKPlayer player, @NonNull FKStream stream, @Nullable FKError error) {
        Log.v(TAG, "onPlayerWillPlay: stream " + stream + " unless error: " + error);
        if (error != null) {
            Toast.makeText(getContext(), "FKError: " + error, Toast.LENGTH_LONG).show();
        }
        hideLoading();
    }

    @Override
    public void onPlayerStateChanged(@NonNull FKPlayer player, FKPlayer.State state) {
        Log.v(TAG, "onPlayerStateChanged: " + state);
        mTextViewState.setText("" + mPlayer.getState());
    }

}

File: app/src/main/java/io/firekast/demo/PlayerFragment.kt
package io.firekast.demo

import android.os.Bundle
import android.support.v4.app.Fragment
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import io.firekast.FKError
import io.firekast.FKPlayer
import io.firekast.FKStream
import kotlinx.android.synthetic.main.fragment_player.*

class PlayerFragment : Fragment(), FKPlayer.Callback {

    private lateinit var player: FKPlayer

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_player, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        editText.setText(gLatestStream?.id)
        progressBar.visibility = View.GONE
        buttonPlay.setOnClickListener(onClickPlayButton)
        buttonPause.setOnClickListener(onClickPauseButton)
        buttonResume.setOnClickListener(onClickResumeButton)

        player = playerView.player
        player.setCallback(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        player.release()
    }

    private val onClickPlayButton = View.OnClickListener {
        progressBar.visibility = View.VISIBLE
        val streamId = editText.text.toString().trim()
        val current = player.currentStream
        if (TextUtils.isEmpty(streamId)) {
            editText.error = "StreamId needed here."
        } else if (current == null || current.id != streamId) {
            // no stream playing yet or wants to play another stream
            val stream = FKStream.newEmptyInstance(streamId)
            player.play(stream)
        } else {
            // restart the stream from the beginning
            player.play(current)
        }
    }

    private val onClickPauseButton = View.OnClickListener {
        player.pause()
    }

    private val onClickResumeButton = View.OnClickListener {
        player.resume()
    }

    // -----
    // FKPlayerCallback
    // -----

    override fun onPlayerWillPlay(stream: FKStream, error: FKError?) {
        progressBar.visibility = View.GONE
        error?.let {
            Toast.makeText(this.context, "Error: $it", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onPlayerStateChanged(state: FKPlayer.State) {
        textViewState.text = "$state"
    }
}